app/uk/gov/hmrc/tai/connectors/CacheConnector.scala

```diff
@@ -94,138 +79,87 @@ class CacheConnector @Inject()(
     } else {
       Json.toJson(data)
     }
-    cacheRepository.createOrUpdate(cacheId.value, key, jsonData).map(_ => data)
+    taiCacheRepository.save(cacheId.value)(key, jsonData).map(_ => data)
   }
 
   def find[T](cacheId: CacheId, key: String = defaultKey)(implicit reads: Reads[T]): Future[Option[T]] =
     if (mongoConfig.mongoEncryptionEnabled) {
       val jsonDecryptor = new JsonDecryptor[T]()
-      cacheRepository.findById(cacheId.value) map {
+      taiCacheRepository.findById(cacheId.value) map {
         case Some(cache) =>
-          cache.data flatMap { json =>
-            (json \ key).validateOpt[Protected[T]](jsonDecryptor).asOpt.flatten.map(_.decryptedValue)
-          }
-        case None => {
-          None
-        }
+          (cache.data \ key).validateOpt[Protected[T]](jsonDecryptor).asOpt.flatten.map(_.decryptedValue)
+        case None => None
       }
     } recover {
       case JsResultException(_) => None
     } else {
-      cacheRepository.findById(cacheId.value) map {
-        case Some(cache) =>
-          cache.data flatMap { json =>
-            (json \ key).validateOpt[T].asOpt.flatten
-          }
-        case None => {
-          None
-        }
+      taiCacheRepository.findById(cacheId.value) map {
```

Comment:
nitpick: In some methods u use OptionT and in some not

Reply:
Agreed. Added to refactoring across TAI

 https://jira.tools.tax.service.gov.uk/browse/PAYEE-944